doc/compiler/aot/Diagnostics.md

```diff
@@ -0,0 +1,420 @@
+conservative. When trying to allow certain optimizations, if not all
+areas are updated to have a consistent set of assumptions, it can lead
+to problems. Often, bugs of this nature are easier to deal with as
+they will also occur in the compile run.
+
+# Debugging Strategies
+Much of the strategies one uses to narrow down the problematic method in
+a JIT compile apply here. However, there are some additional options
+to know about
+- `limit=`, `exclude=`, `limitFile=` apply only to compilations. It is
+worth noting that this will limit both JIT **and** AOT compilations
+regardless of whether it's set on `-Xjit` or `-Xaot`.
+- `loadLimit=`, `loadExclude=`, `loadLimitFile=` apply only to loads;
+these can only be set on `-Xaot`.
+
+Once the problem method is found, one should try and see if the problem
+narrows down to one of the three common problems above.
+
+A missing or incorrect relocation is relatively straightforward if one
+can find a location in the code that has an invalid address;
+cross-referencing with the RT Log can confirm whether the relocation
+exists or was applied properly. An example of such a bug can be found
+[here](https://github.com/eclipse-openj9/openj9/issues/14300). Similarly,
+dealing with invalid code is also often straightforward,
+especially if the problem occurs in the compile run, as it boils down to
+the usual strategies required to debug problems in a JIT compilation.
+As this often occurs development, there is no existing issue to point
+to.
+
+A missing or incorrect validation is unfortunately rarely straightforward.
+Often it involves diving deep into the core dump and the trace logs to
+understand exactly where things go wrong. Once a likely culprit is
+found, the logs can then be cross-referenced to verify whether
+a validation existed or was incorrect. An example of such a bug can be
+found [here](https://github.com/eclipse-openj9/openj9/issues/9710).
+
+# Other related information in the RTLog
+The RT Log also contains information printed by the `TR_J9SharedCache`
+object, which is used by the compiler to interface with the SCC. If the
+RT Log is enabled in the compile run, it will contain information
+such as:
+```
+TR_J9SC:rememberClass class 000000000004E700 romClass 00007F607D477448 com/ibm/jit/JITHelpers
+TR_J9SC:        key created: 00003c89
+TR_J9SC:        creating chain now: 1 + 1 + 1 superclasses + 0 interfaces
+TR_J9SC:                Chain 00007F60B5BB94B0 store chainLength 24
+TR_J9SC:                Chain 00007F60B5BB94B8 storing romclass 00007F607D477448 (com/ibm/jit/JITHelpers) offset 247952
+TR_J9SC:                writeClassesToChain:
+TR_J9SC:                Chain 00007F60B5BB94C0 storing romclass 00007F607D459000 (java/lang/Object) offset 0
+TR_J9SC:                writeInterfacesToChain:
+TR_J9SC:                fillInClassChain returning true
+TR_J9SC:        stored data, chain at 00007F608E8060D0
+```
+This logs the storing the Class Chain, which is part of the any
```

Comment:
couple of typos (extra words here) I think

Reply:
Updated in force push.